### DISCUSSION TITLE
Next Cloudinary upload button works perfectly fine on local host, but not in the Vercel preview.

### DISCUSSION THREAD

**Discussion by alexander-bick** [neuefische/questions#670] [Web] [answered]
Hi everyone,

I am stuck on a problem regarding the functionality of my Next Cloudinary image upload, which works on my local host without any issues, but wouldn´t work in the Vercel preview. The Chrome developer tool shows the following errors when clicking the upload button:

<img width="2044" alt="Bildschirmfoto 2023-08-16 um 11 19 27" src="https://github.com/neuefische/questions/assets/128151050/5721ffd8-f647-4f74-87ff-452d6d4d645b">

I tried to use the use-local-storage-state package already, but it did produce other errors as well. As I couldn´t find any solution to my problem yet, I would be very thankful for any hint to solve the trouble!
This is my actual [PR](https://github.com/alexander-bick/capstone-project/pull/6).

Thanks for checking my problem 🙏🏼

**Comment by ThomasF85** [comment]
I am getting no error when I check the deployment. However the upload is not working and is stuck in loading forever.
Have you tried deleting your localstorage?
Have you set the environment variable in Vercel?

**Reply by alexander-bick** [reply]
The hint about the environment variable was very helpful - just had to change the Cloudinary upload preset as well. Thanks a lot for you help 🙏🏼🙂

**Comment by Kheirah** [comment]
Another detail to look at is the `ml_default` as uploadPreset.

When you first create an account at cloudinary, the default preset is called "ml_default" and its value is set to "signed".
Don't use that default preset. You need to _**create a new preset**_ at this point.

- in cloudinary go to settings
- under "Upload" find the "Upload presets" section
- and click the link "Add upload preset"
- change the "Signing Mode" to "Unsigned" and hit "Save"

You're almost done.
Take the newly generated upload preset name (it's something cryptic like "a5vk3fh") and use that value as the `uploadPreset` value for your `CldUploadButton`.

Push your changes and see if your image upload works on Vercel.

**Reply by alexander-bick** [reply, marked as answer]
It works now - thank you so much for your help Ernst 🙏🏼🙂
